src/main/java/core/basesyntax/parser/FruitParser.java

```
@@ -0,0 +1,21 @@
+package core.basesyntax.parser;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Arrays;
+
+public class FruitParser {
+    public static FruitTransaction parse(String fruitInfo) {
```

This method is missing crucial validation for the input string `fruitInfo`. The task requires handling incorrect data. You should check for:

1.  `null` input.
2.  Incorrect format (e.g., not enough parts after splitting by comma), which would cause an `ArrayIndexOutOfBoundsException`.
3.  Negative quantity, as the quantity of fruit cannot be less than zero.

Failing to handle these cases can cause the application to crash. Please add checks and throw a `RuntimeException` with an informative message for each invalid case.